doc/source/whatsnew/v1.0.0.rst

```diff
@@ -1259,6 +1259,7 @@ ExtensionArray
 - Bug in :class:`arrays.PandasArray` when setting a scalar string (:issue:`28118`, :issue:`28150`).
 - Bug where nullable integers could not be compared to strings (:issue:`28930`)
 - Bug where :class:`DataFrame` constructor raised ``ValueError`` with list-like data and ``dtype`` specified (:issue:`30280`)
+- Bug where :class:`Series` set to scalar extension type was considered an object type rather than the extension type (:issue:`34832`)
```

Comment:
since we had to change some other tests, I think we need to break this to a new section and show the changes from before. E.g. construction of a multi-column df now is object if we don't have unform datetimes? (we need to be very clear what is the change here since we had to change some tests)

Reply:
Realized that I added this to the wrong file, so I'm moving this addition to `v1.1.0.rst`. But can you clarify what you would like me to do here? I'm not totally sure based on your comment.

I think that this line correctly describes the change. Are you asking to also include an example, such as

```
# This used to create an object type column, now it creates a Period type column
DataFrame(index=[0, 1], columns=["a"], data=pd.Period("2020-01"))
```

The example you gave, where a datetime column has multiple different timezones, this was always an object column.

Reply:
ok tell you what. let's create an issue to refactor this (frame constructor). That's what I mainly have a problem with, we are adding if/then ALL over the place for extension types rather than a proper refactor.

So ok with merging this (just one small change on the naming in the tests). And please create an issue (and if you want / can refactor would be great) as a followup.